### PR TITLE
Build v 2.1.1:

### PR DESCRIPTION
**Pull request by akabanovs**
- Added two new patches needed for building.
 - Reset build number to 0.
 - Updated build section with patch.
 - Update host section with wheel.
 - Updated test section of the recipe.
 - Updated 'about' section with some additional info.
 - Added local cbc, although might be redundant.

### CODE REVIEW

**Comment by akabanovs** [comment]
If you review this, can you please suggest on the following:

It seems like psutil is only used in tests, but the upstream specifies it as a required package for build and run. 
https://github.com/search?q=repo%3Anmslib%2Fnmslib+psutil&type=code

It seems like pybind11 is only required for build but the upstream also lists it as a runtime dependency.
https://github.com/search?q=repo%3Anmslib%2Fnmslib+pybind&type=code

I have kept the dependencies as they are suggested in the upstream

testKnnQueryBatch seems to randomly fail on ppc64l. I have deactivated it.

**Review by skupr-anaconda — APPROVED** [review]
LGTM

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+macos_min_version:              # [osx]
+  - 10.7                        # [osx]
```
Comment (boldorider4) [inline recipe/conda_build_config.yaml:2]: Why are we suddenly supporting osx < 10.9?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -18,28 +17,39 @@ source:
     # into various problems around ABIs of numpy/C++ etc.
     # Hopefully, we can provide more specific builds per CPU architecture
     # in the future, see conda-forge/conda-forge.github.io#1261
-    - patches/0002-silence-log-that-tells-people-to-install-from-pip.patch
+    - patches/0001-silence-log-that-tells-people-to-install-from-pip.patch
```
Comment (boldorider4) [inline recipe/meta.yaml:20]: I once heard (and did this myself) that it's always a good idea to be eloquent about the origin of the patches. If the comment above applies, ignore this comment. I would also reference the PRs or GH issues that originate the patches in the PR description, but maybe that's just me.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -18,28 +17,39 @@ source:
...
+    # upstream CI need quotes escaped, we (seems to be an issue on win64 only) need them unescaped
+    - patches/0006-do-not-escape-quotes-for-VERSION_INFO.patch  # [win]
 
 build:
-  number: 2
-  script: {{ PYTHON }} -m pip install ./python_bindings -vv
+  number: 0
+  script: {{ PYTHON }} -m pip install ./python_bindings --no-deps --no-build-isolation -vv
+  missing_dso_whitelist:  # [s390x]
+    - '$RPATH/ld64.so.1'  # [s390x]
```
Comment (boldorider4) [inline recipe/meta.yaml:36]: I don't think $RPATH is necessary, since it is going to be searched by default.

Comment (akabanovs) [reply recipe/meta.yaml:36]: To be honest I just copied what had been done for other packages, to be consistent. Most of them use $RPATH/ld64.so.1:

aggregate find ./* -type f -name 'meta.yaml' -exec grep -H -E 'ld64.so.1' {} \;

./abseil-cpp-feedstock/recipe/meta.yaml:    - '$RPATH/ld64.so.1'  # [s390x]
./boost-1.73-feedstock/recipe/meta.yaml:        - '$RPATH/ld64.so.1'  # [s390x]
./capnproto-feedstock/recipe/meta.yaml:    - $RPATH/ld64.so.1    # [linux]
./cargo-bundle-licenses-feedstock/recipe/meta.yaml:    - $RPATH/ld64.so.1  # [s390x]
./cffi-feedstock/recipe/meta.yaml:    - $RPATH/ld64.so.1  # [s390x]
./cryptography-feedstock/recipe/meta.yaml:    - $RPATH/ld64.so.1    # [s390x]
./cython-blis-feedstock/recipe/meta.yaml:    - $RPATH/ld64.so.1    # [linux and s390x]
./gdbm-feedstock/recipe/meta.yaml:    - $RPATH/ld64.so.1    # [linux and s390x]

File: recipe/meta.yaml [hunk 4]
```diff
@@ -18,28 +17,39 @@ source:
     # into various problems around ABIs of numpy/C++ etc.
     # Hopefully, we can provide more specific builds per CPU architecture
     # in the future, see conda-forge/conda-forge.github.io#1261
-    - patches/0002-silence-log-that-tells-people-to-install-from-pip.patch
+    - patches/0001-silence-log-that-tells-people-to-install-from-pip.patch
     # backport nmslib/nmslib#488 for compatibility with newer pybind11
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml]: I think this comment is not needed anymore

File: recipe/meta.yaml (hunk 3 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:36]: yes, I saw the same in other feedstock, I think that is OK

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (akabanovs) [reply recipe/conda_build_config.yaml:2]: These are specified in setup.py:
https://github.com/nmslib/nmslib/blob/v2.1.1/python_bindings/setup.py#L132#L134

I think even without the local cbc file, these will applied.

Comment (boldorider4) [reply recipe/conda_build_config.yaml:2]: so if without cbc they will be applied, why do you need this cbc?

File: recipe/meta.yaml (hunk 4 above)
Comment (boldorider4) [reply recipe/meta.yaml]: indeed

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (akabanovs) [reply recipe/conda_build_config.yaml:2]: For better visibility. It does not hurt.

File: recipe/meta.yaml (hunk 4 above)
Comment (akabanovs) [reply recipe/meta.yaml]: I will remove it upon merging. Don't want to run the prefect just to rebuild without this comment.

File: recipe/meta.yaml (hunk 3 above)
Comment (boldorider4) [reply recipe/meta.yaml:36]: Can you try without $RPATH and see if it builds? I think if RPATH is searched by default, even though other recipes may include it, it should not be included (and we should set the right example ;) )

File: recipe/meta.yaml (hunk 2 above)
Comment (akabanovs) [reply recipe/meta.yaml:20]: I am unable to find all the PRs or GH issues that mention the fixes covered by patches. Four patches were already there, added by CF. I was unable to build without these.

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (boldorider4) [reply recipe/conda_build_config.yaml:2]: so if without cbc they will be applied, why do you need this cbc?

Comment (boldorider4) [reply recipe/conda_build_config.yaml:2]: Ok...maybe it will spark a question in the future

File: recipe/meta.yaml (hunk 3 above)
Comment (akabanovs) [reply recipe/meta.yaml:36]: It fails:
https://cloud.prefect.io/anaconda/flow-run/c4fd1791-85f0-4e14-b594-12ad59bd7251?logs

We should use either '$RPATH/ld64.so.1' or '*/ld64.so.1'

Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:36]: RPATH has to be here always because the feedstock build is free to set up RPATH in any location

**Review by boldorider4 — APPROVED** [review]

**Pull request merged**